### DISCUSSION TITLE
[HW Accel Support]: ARC A310 Support? Working Setup Questions.

### DISCUSSION THREAD

**Discussion by moogoos** [blakeblackshear/frigate#14799] [Hardware Acceleration Support] [answered]
### Describe the problem you are having

I dont have a problem, necessarily. Curious about my upgraded setup.

I upgraded from an Intell 14500 Q670 platform, using an M.2 Coral card which worked pretty well and iGPU. I've had mostly success with A+E and M.2 Coral cards in a couple builds. I recently decided to make some changes, and consolidate to the newer z890 Intel Ultra Core platform and see if it would work. Overall the consolidation went well. The only issue I had was with the M.2 device. It has some problem potentially with power management. I keep getting a device inaccessible and some other errors so a driver or BIOS problem perhaps. So I became stuck. I could have reverted to the Tensor setup I had a config for and hardware for but want to keep the power down as much as possible, which is why I prefer Coral with iGPU.

My setup is Proxmox >> VM >> Docker with PCIe passthrough.

I grabbed a sparkle A310 card to play around with for transcoding and decided to just move my containers to the VM that has this card passed to it as passthrough. It shows up as /dev/dri is passed to the container from the VM. The VM has the IOMMU group passed to it from the host, so the A310 card is passed to the VM. That is what makes /dev/dri and card0 card1 and renderD128 available, I think. The iGPU is NOT passed to the VM.

I then switch to use CPU type detection instead of coral in my frigate config.

`detectors:
  cpu1:
    type: cpu
    num_threads: 3
`
docker-compose device pass through

`
- /dev/dri/renderD128 # for intel hwaccel, needs to be updated for your hardware
`

So if im understanding this correctly, this should then be using the A310 for detect.

Either way the setup is working extremely well. It's just as fluid and responsive as it was using the Coral card,  I was going to maybe get the USB Coral but I'm not a fan of USB and at about $80 bucks I really didn't want another card even though it probably would have worked. What im trying to understand is how can I confirm that the GPU is working detect? I'm running 8 detect sub streams. I just dont see a reason to use a coral card now at all?:

`    width: 1280
      height: 720
      fps: 5
`

detected inference is about 24ms

`
Vainfo Output
Return Code: 0

Process Output:

vainfo: VA-API version: 1.22 (libva 2.10.0)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 24.3.3 ()
vainfo: Supported profile and entrypoints
VAProfileNone : VAEntrypointVideoProc
VAProfileNone : VAEntrypointStats
VAProfileMPEG2Simple : VAEntrypointVLD
VAProfileMPEG2Main : VAEntrypointVLD
VAProfileH264Main : VAEntrypointVLD
VAProfileH264Main : VAEntrypointEncSliceLP
VAProfileH264High : VAEntrypointVLD
VAProfileH264High : VAEntrypointEncSliceLP
VAProfileJPEGBaseline : VAEntrypointVLD
VAProfileJPEGBaseline : VAEntrypointEncPicture
VAProfileH264ConstrainedBaseline: VAEntrypointVLD
VAProfileH264ConstrainedBaseline: VAEntrypointEncSliceLP
VAProfileHEVCMain : VAEntrypointVLD
VAProfileHEVCMain : VAEntrypointEncSliceLP
VAProfileHEVCMain10 : VAEntrypointVLD
VAProfileHEVCMain10 : VAEntrypointEncSliceLP
VAProfileVP9Profile0 : VAEntrypointVLD
VAProfileVP9Profile0 : VAEntrypointEncSliceLP
VAProfileVP9Profile1 : VAEntrypointVLD
VAProfileVP9Profile1 : VAEntrypointEncSliceLP
VAProfileVP9Profile2 : VAEntrypointVLD
VAProfileVP9Profile2 : VAEntrypointEncSliceLP
VAProfileVP9Profile3 : VAEntrypointVLD
VAProfileVP9Profile3 : VAEntrypointEncSliceLP
VAProfileHEVCMain12 : VAEntrypointVLD
VAProfileHEVCMain422_10 : VAEntrypointVLD
VAProfileHEVCMain422_10 : VAEntrypointEncSliceLP
VAProfileHEVCMain422_12 : VAEntrypointVLD
VAProfileHEVCMain444 : VAEntrypointVLD
VAProfileHEVCMain444 : VAEntrypointEncSliceLP
VAProfileHEVCMain444_10 : VAEntrypointVLD
VAProfileHEVCMain444_10 : VAEntrypointEncSliceLP
VAProfileHEVCMain444_12 : VAEntrypointVLD
VAProfileHEVCSccMain : VAEntrypointVLD
VAProfileHEVCSccMain : VAEntrypointEncSliceLP
VAProfileHEVCSccMain10 : VAEntrypointVLD
VAProfileHEVCSccMain10 : VAEntrypointEncSliceLP
VAProfileHEVCSccMain444 : VAEntrypointVLD
VAProfileHEVCSccMain444 : VAEntrypointEncSliceLP
VAProfileAV1Profile0 : VAEntrypointVLD
VAProfileAV1Profile0 : VAEntrypointEncSliceLP
VAProfileHEVCSccMain444_10 : VAEntrypointVLD
VAProfileHEVCSccMain444_10 : VAEntrypointEncSliceLP
`

### Version

15-beta

### Frigate config file

```yaml
NA
```

### docker-compose file or Docker CLI command

```yaml
NA
```

### Relevant Frigate log output

```shell
NA
```

### Relevant go2rtc log output

```shell
NA
```

### FFprobe output from your camera

```shell
NA
```

### Install method

Docker Compose

### Object Detector

Other

### Network connection

Wired

### Camera make and model

NA

### Screenshots of the Frigate UI's System metrics pages

NA

**Comment by NickM-27** [comment]
using the `cpu` detector does not use the intel GPU, you need openvino for that

**Reply by moogoos** [reply]
I see. So if it's just hwaccell is that for the video streams then that the GPU is using? The CPU seems to be doing a good job for detect then even in a VM. What would the fallout be here that I can't see?

**Reply by NickM-27** [reply]
Let's see a screenshot of the system metrics

**Reply by moogoos** [reply]
![Screenshot 2024-11-04 at 6 43 32 PM](https://github.com/user-attachments/assets/dbb89675-fd24-4ec3-89b8-6f42886e1c78)

**Reply by NickM-27** [reply]
That's not really running well, CPU usage is very high

**Reply by moogoos** [reply]
> That's not really running well, CPU usage is very high

It's always been high for me even with coral. Isn't a big part of detect CPU only?

**Reply by NickM-27** [reply]
No, if it's running properly it should be much less than what yours is showing, and it would be if you used the GPU

**Reply by moogoos** [reply]
Ok so it's working but just not efficiently? What option should I go with then in this setup? I guess i'll grab a coral USB if necessary or maybe ill try openvino. What would you suggest? I would assume then it's not using the GPU at all? i'm confused as i'm passing the renderD128 device which should be the card.

coarl usb
openvino

**Reply by NickM-27** [reply]
Either would with fine, but openvino would be simpler to setup on your environment

**Reply by moogoos** [reply]
kk so openvino should use the GPU then?

**Reply by NickM-27** [reply]
Yes

**Reply by moogoos** [reply]
alright i switched over. Will run it for a day and compare. Thanks for the help!

**Reply by moogoos** [reply]
What is the significance of width and height here?

model:
  width: 300
  height: 300

for openvino config?

**Reply by NickM-27** [reply]
The size of the model

**Reply by moogoos** [reply]
ill ask a seperate question on that width height settings.

**Reply by moogoos** [reply, marked as answer]
openvino looks MUCH better and the inference time is extremely reasonable.

![Screenshot 2024-11-05 at 1 13 52 PM](https://github.com/user-attachments/assets/bb419fe1-30ee-4d68-aaf5-e00249712aaf)
